request/request.go

```diff
@@ -208,6 +208,10 @@ func (r *Request) SizeAndDo(m *dns.Msg) bool {
 		mo.SetUDPSize(o.UDPSize())
 		mo.Hdr.Ttl &= 0xff00 // clear flags
 
+		if len(o.Option) > 0 {
+			o.Option = supportedOptions(o.Option)
```

Comment:
Perhaps it should be smth like
```
mo.Option = append(mo.Option, supportedOptions(o.Option)...)
```
i.e. we should not remove (overwrite) existing options from response which might come from upstream server.
Also, we likely need to add check for duplicates in the code above, i.e. don't add option again if it already presents in response OPT record.

Reply:
no, that will not work, because we whip the entire opt record anyway in forward and proxy right now

Reply:
@miekg, please check at least if `o.Option =` should be `mo.Option =`

Seems, in your solution you remove unknown options from query message, shouldn't you remove them from response?

See also line 23
```
m.Extra = append(m.Extra, o)
```
which is called only if response doesn't have OPT record

Reply:
I'm talking not about removing entire OPT RR, but about filtering individual options which you do in `supportedOptions()`. 
You take the options list from query OPT record, filter this list and assign back to **query** OPT. This doen't affect **response** OPT record.

Reply:
... uh yes?

I don't understand what you're getting at: 
* is this something we need to do? 
* did we do this before and not anymore?
* Is there an actual bug in the PR?

Reply:
I meant to say that the code
https://github.com/coredns/coredns/blob/a2ad651405f1febaec4ffbd677f9ac190dada1a7/request/request.go#L211-L213
doesn't make any sense here in the block
```
if mo := m.IsEdns0(); mo != nil {
```
because we have `return` at the end of this block, and the modified object `o` is not used before `return`

For comparison, the similar code at
https://github.com/coredns/coredns/blob/a2ad651405f1febaec4ffbd677f9ac190dada1a7/request/request.go#L226-L228
may have sense, because the modified `o` object is used in line
https://github.com/coredns/coredns/blob/a2ad651405f1febaec4ffbd677f9ac190dada1a7/request/request.go#L233
before `return` statement

So, I just asked you to check if this code at lines 211-213 is really needed, or should be modified to make sense

Reply:
you have a point